integration-test/run.sh

```diff
@@ -0,0 +1,179 @@
+popd
+
+# Build the variants
+unset idx
+if [ ! -d spark ]; then
+  git clone "${SPARK_REPO}"
+fi
+
+
+for branch in "${SPARK_BRANCHES_ARRAY[@]}"
+do
+  # Copy so we don't have to clean build everytime if we're recycling the run dir.
+  if [ ! -d "spark-${branch}" ]; then
+    cp -af spark "spark-${branch}"
+  fi
+  pushd "spark-${branch}"
+  (git checkout "${branch}" && \
+#   git pull && \
+#   ./build/mvn -Pkubernetes -Phadoop-aws compile package -DskipTests && \
+  (./bin/docker-image-tool.sh -r "${CONTAINER_PREFIX}" -t "${TAG}-${branch}" -b java_image_tag=11-jre-slim -X -p resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile build || ( \
+   ./bin/docker-image-tool.sh -r "${CONTAINER_PREFIX}" -t "${TAG}-${branch}" -b java_image_tag=11-jre-slim  -p resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile build && \
+   ./bin/docker-image-tool.sh -r "${CONTAINER_PREFIX}" -t "${TAG}-${branch}" push)) && \
+  SPARK_TAGS+=("${TAG}-${branch}") \
+  ) || echo "Spark branch ${branch} failed at $(git log -n 5)"
+  popd
+done
+
+if [ ! -d flink-docker ]; then
+  git clone git@github.com:apache/flink-docker.git
+fi
+pushd "flink-docker/${FLINK_VERSION}/scala_2.12-java11-debian"
+docker build . -t "${CONTAINER_PREFIX}/flink:${TAG}"
+popd
+
+echo "Building support tools (TPCDS, etc.)"
+pushd "${INTEGRATION_RUN_DIR}"
+if [ ! -d spark-tpcds-datagen ]; then
+  git clone git@github.com:maropu/spark-tpcds-datagen.git
+fi
+pushd spark-tpcds-datagen
+#./build/mvn package -DskipTests
+popd
+
+echo "Build the containers with Iceberg & tools present"
+rm -rf iceberg
+cp -af ${ICEBERG_DIR} ./iceberg
+
+for SPARK_TAG in "${SPARK_TAGS[@]}"
+do
+  docker buildx build . -f "${INTEGRATION_DIR}/containers/spark/Dockerfile" -t "${CONTAINER_PREFIX}/iceberg-spark:${SPARK_TAG}" --build-arg base="${CONTAINER_PREFIX}/spark-py:${SPARK_TAG}" --push ${DOCKER_EXTRA}
+done
+
+docker buildx build . -f "${INTEGRATION_DIR}/containers/flink/Dockerfile" -t "${CONTAINER_PREFIX}/iceberg-flink:${TAG}" --build-arg base="${CONTAINER_PREFIX}/flink:${TAG}" --push ${DOCKER_EXTRA}
+
+popd
+
+if [ "$SKIP_MINIO" != "true" ]; then
+  echo "Deploy minio"
+  helm repo add minio https://helm.min.io/
+  helm repo update
+  deployment=minio-iceberg-part
+  helm status --namespace "${TEST_NS}" "${deployment}" || helm install --namespace "${TEST_NS}"  --set accessKey=myaccesskey,secretKey=mysecretkey --set image.repository="${MINIO_REPO}" --set image.tag="${MINIO_TAG}" --set defaultBucket.enabled=true  --set persistence.enabled=false --set resources.requests.memory=10Gi --set resources.limits.memory=12Gi --set podLabels."sdr\.appname"="minio" "${deployment}" minio/minio
+  export S3_ACCESS_KEY=myaccesskey
+  export S3_SECRET_KEY=mysecretkey
+  export S3_ENDPOINT=${deployment}.${TEST_NS}.svc.cluster.local
+  export S3_ROOT="s3a://bucket"
+fi
```

Comment:
Currently, if we skip minio, S3_ACCESS_KEY, S3_ENDPOINT, and friends don't get exported unless they're already in the environment or they come from the `extras.sh`.

Definitely we shouldn't be hardcoding real secrets, but pointing this out so I don't forget as it tripped me up at least once.